Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -53,12 +53,13 @@ const double THRESHOLD = 1E-6;
 
 /** removes zeros from converged results
 * @param ws :: [input] The input workspace with zeros
-* @param maxIt :: [input] The max number of iteration this alg used
+* @param itCount [input] The number of iterations this alg used for each
+* spectrum
 * @param yLabel :: [input] y-label to use for returned ws
 * @return : ws cut down in lenght to maxIt
 */
-MatrixWorkspace_sptr removeZeros(const MatrixWorkspace_sptr &ws,
-                                 const size_t maxIt,
+MatrixWorkspace_sptr removeZeros(MatrixWorkspace_sptr &ws,
+                                 const std::vector<size_t> &itCount,
                                  const ::std::string yLabel) {
 
   ws->setYUnitLabel(yLabel);
```

Comment:
I know the code below regarding the units was not changed here but it looks odd. "Number of Iterations" is not a valid unit identifier so this will always go through the exception. I would suggest removing the `try` and just create the "Label" unit as in the catch case.

Reply:
Please comment on code above. That is what GitHub displays in the pull request.

Reply:
@KarlPalmen Sorry for not being clear - the way the code is displayed makes it hard to comment on lines that have not changed. I was referring to [lines 66-75](https://github.com/mantidproject/mantid/blob/ef9307e26a46c626414ff0b43100d47e808a9452/Framework/Algorithms/src/MaxEnt.cpp#L66) with the `try...catch` around setting the units.